include/seqan3/io/sam_file/detail/format_sam_base.hpp

```diff
@@ -481,9 +480,9 @@ inline void format_sam_base::read_header(stream_view_type && stream_view,
                         else
                             throw format_error{std::string{"Illegal SAM header tag: S"} + *it};
                     }
-                    else if (!is_char<'G'>(*it))                        // GO (grouping) tag
+                    else                                                // GO (grouping) tag, pb (PacBio) version, etc
```

Comment:
You changed some logic here.

* `GO` tags will now emit warnings even though they shouldn't
* unknown tags are passed to `parse_tag_value(*who);` which might not be correct?
* * Maybe we need to skip the unknown tag with `take_until_predicate` but I am not sure.

Reply:
> - GO tags will now emit warnings even though they shouldn't

But we didn't do anything with the GO tag before, did we? So I thought I would include GO in the warning. But I can also undo that.

> - unknown tags are passed to parse_tag_value(*who); which might not be correct?
        - Maybe we need to skip the unknown tag with take_until_predicate but I am not sure.



`take_until_predicate` just skips everything until the next day? Well, since we don't support them, that sounds like the right way to go. Maybe @eldariont has a hint if it is worth to save them.

Reply:
We can also think about making a specific distinction between upper case and lower case. Because lower case comes from the user and upper case should be in the specification, if I understood correctly.

Reply:
> But we didn't do anything with the GO tag before, did we? So I thought I would include GO in the warning. But I can also undo that.

It was passed to `parse_tag_value(*who);` before.

> `take_until_predicate` just skips everything until the next day? 

`take_until_predicate(is_char<'\t'>)` or similar syntax should work

> Well, since we don't support them, that sounds like the right way to go. Maybe @eldariont has a hint if it is worth to save them.

We should support them at some point, but this would be a new feature (and would need to use the sam_tag_dictionary). For now, we should just ignore them.

Reply:
> We can also think about making a specific distinction between uppercase and lowercase. Because lower case comes from the user and upper case should be in the specification, if I understood correctly.

It's not forbidden to do user-defined uppercase tags, it's just that lowercase are reserved for this, and it's best practice to use lowercase.
I wouldn't make a distinction between lower/uppercase

Reply:
Okay, than I will skip them with your given syntax and open an issue for supporting them later.

Reply:
I have never used any of the HD tags except SO (sorting order). Nevertheless, I agree with @eseiler that seqAn should not emit a warning for tags defined in the specs, like the GO tag.